### ISSUE TITLE
Complex parametrized query takes an abnormal amount of time to execute

### ISSUE THREAD

**Issue by juanpardo** [crate/crate#15997]
### CrateDB version

5.5.2

### CrateDB setup information

Number of nodes: 3
Heap size: 1879048192
JAVA_OPTS: -Dcom.sun.management.jmxremote.port=6666 -Dcom.sun.management.jmxremote.ssl=false
        -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.local.only=false
        -Dcom.sun.management.jmxremote.rmi.port=6666 -Djava.rmi.server.hostname=127.0.0.1
        -javaagent:/var/lib/crate/crate-jmx-exporter-1.0.0.jar=7071 -XX:+HeapDumpOnOutOfMemoryError
        -XX:HeapDumpPath=/resource/heapdump -Dlog4j2.formatMsgNoLookups=true

### Problem description

The query provided takes several minutes to be parsed and ends up failing with the following error:

```
SQL Error [XX000]: ERROR: Couldn't create execution plan from logical plan because of: java.lang.ArrayIndexOutOfBoundsException: Index 260 out of bounds for length 0:
```
This error could be an artifact of trying to reproduce the issue locally, though.
The was only reproducible when the query is parametrized.

In addition to that, the query never appeared in the `sys.jobs` nor `sys.jobs_log`. This meant finding the issue was much harder than it could have been. Is this the expected behaviour? If yes, what would be a good way to detect this type of problems instead of using the mentioned tables?

The query:
```
SELECT
    doc.categories.id,
    doc.categories.times,
    doc.categories.name,
    doc.categories.brand,
    doc.categories.store_id,
    doc.categories.soft_deleted,
    doc.products.id,
    doc.products.name,
    doc.products.times,
    doc.products.color,
    doc.products.size_a,
    doc.products.size_b,
    doc.products.category_id,
    doc.products.size_c,
    doc.products.weight,
    doc.products.description,
    doc.products.a,
    doc.products.b,
    doc.products.c,
    doc.products.d,
    doc.products.e,
    doc.products.f,
    doc.products.g,
    doc.products.h,
    doc.products.status,
    doc.products.i,
    doc.products.j,
    doc.products.k,
    doc.products.l,
    doc.products.soft_deleted
    JOIN doc.categories ON doc.products.category_id = doc.categories.id WHERE doc.products.category_id IN ( SELECT DISTINCT
            doc.categories.id nFROM doc.categories
            JOIN doc.roles ON doc.roles.store_id = doc.categories.store_id WHERE(doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?
                    OR doc.roles.user_id = ?
                    AND doc.roles.value = ?
                    AND doc.roles.store_id = ?)
            AND (doc.categories.soft_deleted = FALSE
                OR doc.categories.soft_deleted IS NULL))
    AND (doc.categories.soft_deleted = FALSE
        OR doc.categories.soft_deleted IS NULL)
    AND (doc.products.soft_deleted = FALSE
        OR doc.products.soft_deleted IS NULL)
ORDER BY
    doc.products.times['date_created'] "
```

### Steps to Reproduce

Please contact me to provide a reproducibility path.

### Actual Result

Our cluster started consuming 100% of CPU for a long time, until the query failed, which took several minutes.
The `sys.jobs` and `sys.jobs_log` table did not show any query being the root cause for this (the provided query did not appear in them).

JFR analysis led to suspecting a query with a high count of "and/or" expressions in the `WHERE` clause could be causing the issue.

### Expected Result

We would expect the query to appear in the `sys.jobs` and `sys_jobs_log` tables.
In addition to that we would have expected for the queries to either be executed completely or fail fast (due to hitting some limit, perhaps).

**Comment by matriv** [comment]
More info was shared privately by @juanpardo, thx!

**Comment by matriv** [comment]
Simpler reproduction:
```
CREATE TABLE IF NOT EXISTS t (
    "a" TEXT NOT NULL,
    "b" TEXT NOT NULL,
    "c" TEXT NOT NULL,
    PRIMARY KEY ("a", "b", "c")
);
```

query:
```
SELECT * FROM t
WHERE a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
    OR a = ?
    AND b = ?
    AND c = ?
;
```

**Comment by matriv** [comment]
The issue with ugly `ArrayIndexOutOfBounds` error can simply be reproduced with:
```
SELECT * FROM t
WHERE a = ?
    AND b = ?
    AND c = ?
```
since the where clause contains all columns of the PK and a `Get` plan is used, so `SubQueryAndParamBinder` is not called, which already handles the situation by throwing
a user friendly error messsage.

**Comment by matriv** [comment]
@juanpardo Thx a lot for reporting this issue.
The part regarding the ugly exception has been fixed and will be available with the next hotfix release.

Additionally, I discovered the root cause of the slow process of the query (during the optimization phase) and opened a ticket: https://github.com/crate/crate/issues/16066 to also address this. We'll discuss if this will be treated as a bug or (most probably) as an improvement.
